llvm/lib/Transforms/Utils/SimplifyLibCalls.cpp

```diff
@@ -195,6 +195,32 @@ static void annotateNonNullAndDereferenceable(CallInst *CI, ArrayRef<unsigned> A
   }
 }
 
+static void mergeAttributes(CallInst *CI, AttributeList Attributes) {
+  // TODO: make this a member function on Instruction?
+  CI->setAttributes(
+      AttributeList::get(CI->getContext(), {CI->getAttributes(), Attributes}));
+}
+
+static MemTransferInst *
+createMemcpyFromStringCopy(IRBuilderBase &B, Value *Dst, MaybeAlign DstAlign,
```

Comment:
Not sure what this function achieves; you have to change every instance to call this function already, why not just add the argument and keep things closer to upstream?

Reply:
I originally did that. At some point I changed it to add the helper function, but I can't remember why. Maybe I didn't like the formatting.